lib/daru/index.rb

```diff
@@ -84,6 +84,66 @@ def [](key, *rest)
       end
     end
 
+    # Returns true if all arguments are either a valid category or position
+    # @param [Array<object>] *indexes categories or positions
+    # @return [true, false]
+    # @example
+    #   idx.respond? :a, 2
+    #   # => true
+    #   idx.respond? 3
+    #   # => false
+    def respond? *indexes
+      indexes.all? { |i| to_a.include?(i) || (i.is_a?(Numeric) && i < size) }
+    end
+
+    # Returns positions given indexes or positions
+    # @note If the arugent is both a valid index and a valid position,
+    #   it will treated as valid index
+    # @param [Array<object>] *indexes indexes or positions
+    # @example
+    #   x = Daru::Index.new [:a, :b, :c]
+    #   x.pos :a, 1
+    #   # => [0, 1]
+    def pos *indexes
+      indexes = preprocess_range(indexes.first) if indexes.first.is_a? Range
+
+      if indexes.size == 1
+        self[indexes.first]
+      else
+        indexes.map { |index| by_single_key index }
+      end
+    end
+
+    def subset *indexes
```

Comment:
>  I stand the ground that it would be more reasonable to use bare positions there, as you are already having them.

Yep, thats very reasonable. We now need a way to deal with `MultiIndex` issue.